packages/flutter/lib/src/material/page.dart

```diff
@@ -267,12 +98,12 @@ class MaterialPageRoute<T> extends PageRoute<T> {
   Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> forwardAnimation, Widget child) {
     if (Theme.of(context).platform == TargetPlatform.iOS &&
         Navigator.of(context).userGestureInProgress) {
-      return new _CupertinoPageTransition(
+      return new CupertinoPageTransition(
         animation: new AnimationMean(left: animation, right: forwardAnimation),
         child: child
```

Comment:
Well to be very detailed, there is a single frame flash of the offscreen canvas leaking shadow into the screen. But that's going to go away too as we increase fidelity with the native look (where even the elevation animates from right to left).